zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -830,6 +830,29 @@ function ParagraphCtrl ($scope, $rootScope, $route, $window, $routeParams, $loca
       // autocomplete on 'ctrl+.'
       $scope.editor.commands.bindKey('ctrl-.', 'startAutocomplete')
 
+      // Show autocomplete on tab
+      $scope.editor.commands.addCommand({
+        name: 'tabAutocomplete',
+        bindKey: {
+          win: 'tab',
+          mac: 'tab',
+          sender: 'editor|cli'
+        },
+        exec: function(env, args, request) {
+          let iColumnPosition = $scope.editor.selection.getCursor().column
+
+          // If user has pressed tab on first line char or if editor mode is %md, keep existing behavior
+          // If user has pressed tab anywhere in between and editor mode is not %md, show autocomplete
+          if (iColumnPosition && $scope.paragraph.config.editorMode !== 'ace/mode/markdown') {
```

Comment:
I think python interpreter should use the existing auto completion key.

Reply:
@Leemoonsoo I agree with point 1, I will make the changes in this PR.

Regarding point 2, can we take it up as a separate item with some feedback from the users so that we can make customer centric decision?

Reply:
Okay, 

Then, do you mind If I create a pullrequest to your branch for point 2? I really think it's benefit to developers who develop new interpreter choose completion key without touching front-end code.

Reply:
@Leemoonsoo Sure.